nifi-ignite-extensions/nifi-ignite-processors/src/main/java/pl/touk/nifi/ignite/IgniteBinaryBuilder.java

```diff
@@ -0,0 +1,39 @@
+package pl.touk.nifi.ignite;
+
+import org.apache.ignite.IgniteBinary;
+import org.apache.ignite.binary.BinaryObject;
+import org.apache.ignite.binary.BinaryObjectBuilder;
+import org.apache.nifi.serialization.record.MapRecord;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class IgniteBinaryBuilder {
+
+    private IgniteBinary igniteBinary;
+
+    public IgniteBinaryBuilder(IgniteBinary igniteBinary) {
+        this.igniteBinary = igniteBinary;
+    }
+
+    public BinaryObject fromMapRecord(MapRecord record, String igniteTypeName) {
+        return fromMapRecord(record, igniteTypeName, null);
+    }
+
+    public BinaryObject fromMapRecord(MapRecord record, String igniteTypeName, String[] fieldNamesIncluded) {
+        Map<String, Object> recordFields;
+        if (fieldNamesIncluded == null)
+            recordFields = record.toMap();
+        else {
+            recordFields = record.toMap().entrySet().stream().filter(recordEntry ->
+                    Arrays.stream(fieldNamesIncluded).anyMatch(fieldName ->
+                            recordEntry.getKey().equals(fieldName))
+            ).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        }
```

Comment:
IMHO the function would be much easier to read if you extracted a few functions. i.e. this if/else could be closed in sth like `private Map<String, Object> recordToMap(MapRecord record, String[] fieldNamesIncluded) {` - then you wouldn't need to split Map declaration from assignment.

Also, the filter predicate could be expressed by sth like `private boolean hasAtLeastOneFieldName(Map.Entry<String, Object> recordEntry, String[] fieldNamesIncluded) {`

I'm not too sure about function names I'm suggesting ;)